tests/batching/test_adaptive.py

```diff
@@ -0,0 +1,92 @@
+import asyncio
+
+from mlserver.batching.adaptive import AdaptiveBatcher
+from mlserver.types import InferenceRequest, RequestInput
+from mlserver.model import MLModel
+
+from .conftest import TestRequestSender
+
+
+async def test_batch_requests(
+    adaptive_batcher: AdaptiveBatcher,
+    send_request: TestRequestSender,
+):
+    max_batch_size = adaptive_batcher._max_batch_size
+    sent_requests = await asyncio.gather(
+        *[send_request() for _ in range(max_batch_size)]
+    )
+    batched_requests = [
+        batched_req async for batched_req in adaptive_batcher._batch_requests()
+    ]
+
+    assert len(batched_requests) == 1
+    assert batched_requests[0]._inference_requests == sent_requests
+
+
+async def test_batch_requests_timeout(
```

Comment:
maybe add a note why this is testing timeout and perhaps send another request after timeout and then assert that you end up with two batches?

Reply:
Good point, I'll add that :+1: